src/stores/game.js

```diff
@@ -116,17 +117,36 @@ export const useGameStore = defineStore('game', () => {
   // Stores
   const authStore = useAuthStore();
   const gameHistoryStore = useGameHistoryStore();
+  const route = useRoute();
 
   // Computed (getters)
   const myPNum = computed(() => {
     if (gameHistoryStore.isSpectating) {
+
+      const queryPNum = Number(route.query.pNum);
+      // If valid pNum, return
+      if (queryPNum === 0 || queryPNum === 1) {
+        return queryPNum;
+      }
+
+      const userPNum = players.value.findIndex(({ username }) => username === authStore.username);
+      const userWasPlayer = userPNum !== -1;
+      const gameIsComplete = [ GameStatus.FINISHED, GameStatus.ARCHIVED ].includes(status.value);
+
+      // Check if a complete game is viewed by a player
+      if (userWasPlayer && gameIsComplete) {
+        return userPNum;
+      }
+
       return 0;
     }
     const pNum = players.value.findIndex(({ username }) => username === authStore.username);
     return pNum > -1 ? pNum : null;
```

Comment:
Logic is correct. I'm wondering if we can refactor this to DRY up the variables for looking up the pNum by username in the players array, given that we do that in both spectate and non spectate cases